src/commands.rs

```diff
@@ -758,40 +758,43 @@ where
         .collect::<Vec<Value>>()
 }
 
-/// Sort the paths so higher indices precede lower indices on the same on the same array
-/// And objects with higher hierarchy (closer to the top-level) preceded objects with deeper hierarchy
+/// Sort the paths so higher indices precede lower indices on the same array,
+/// And if one path is a sub-path of the other, then paths with shallower hierarchy (closer to the top-level) precedes paths with deeper hierarchy
 fn prepare_paths_for_deletion(paths: &mut Vec<Vec<String>>) {
-    paths.sort_by(|v1, v2| match (v1.len(), v2.len()) {
-        (l1, l2) if l1 < l2 => Ordering::Less, // Shorter paths before longer paths
-        (l1, l2) if l1 > l2 => Ordering::Greater, // Shorter paths before longer paths
-        _ => v1
-            .iter()
-            .zip(v2.iter())
-            .fold_while(Ordering::Equal, |_acc, (p1, p2)| {
-                let i1 = p1.parse::<usize>();
-                let i2 = p2.parse::<usize>();
-                match (i1, i2) {
-                    (Err(_), Err(_)) => match p1.cmp(p2) {
-                        // String compare
-                        Ordering::Less => Done(Ordering::Less),
-                        Ordering::Equal => Continue(Ordering::Equal),
-                        Ordering::Greater => Done(Ordering::Greater),
-                    },
-                    (Ok(_), Err(_)) => Done(Ordering::Greater), //String before Numeric
-                    (Err(_), Ok(_)) => Done(Ordering::Less),    //String before Numeric
-                    (Ok(i1), Ok(i2)) => {
-                        // Numeric compare - higher indices before lower ones
-                        if i1 < i2 {
-                            Done(Ordering::Greater)
-                        } else if i2 < i1 {
-                            Done(Ordering::Less)
-                        } else {
-                            Continue(Ordering::Equal)
+    paths.sort_by(|v1, v2| {
+        v1.iter()
+            .zip_longest(v2.iter())
+            .fold_while(Ordering::Equal, |_acc, v| {
+                match v {
+                    EitherOrBoth::Left(_) => Done(Ordering::Greater), // Shorter paths before longer paths
+                    EitherOrBoth::Right(_) => Done(Ordering::Less), // Shorter paths before longer paths
+                    EitherOrBoth::Both(p1, p2) => {
```

Comment:
What would have happened before if lenght of the paths are different?

Reply:
Before, we did not compare content of vectors with different lengths.
We only compared content for similar lengths, but this was wrong since a longer path could overlap with a shorter path and must be handled before, as seen in #529